src/services/fileServices/MdPageServices/PageService.ts

```diff
@@ -138,10 +138,6 @@ export class PageService {
           CollectionPageName | SubcollectionPageName,
           NotFoundError
         >((rawPath) => {
-          // NOTE: Only 2 levels of nesting
-          if (rawPath.length > 2) {
-            return errAsync(new NotFoundError())
-          }
```

Comment:
hey, just checking in ya, are we guaranteed that the `length` will be <= 2?

Reply:
this depends on whether the site is actually on isomercms - if they are, then then folders can only be 2 levels deep (pages can either be in a folder or in a subfolder and they can't create another folder within the subfolder)

Reply:
Hmm, if that is the case should we still throw an error if this condition is the `true`? eg. 
```
    // NOTE: Only 2 levels of nesting
    if (rawPath.length > 2) {
       return errAsync(new BaseIsomerError(`Folder/page >2 levels deep, unsupported by IsomerCMS`))
    }
```
rather than not handling it at all?

Reply:
i don't think there's an error thrown here; could i just clarify that what you meant here is to return a more specific error type with a relevant message rather than returning a generic `NotFoundError` which might not be useful?

if that's the case, then yeap, fully agreed (got lazy and didn't do it HAHA) and i'll make the relevant changes so that it returns a `LegacyError`!

Reply:
Hmm correct me if I am wrong, but there is an error thrown `if rawPath.length > 2` at line 160 right? 
To be honest, I am ok this as is (I think returning `NotFoundError()`), but perhaps a quick change that we could do is add some meaningful information about the error? 

Concretely, what do you think about the suggestion:
`return errAsync(new NotFoundError(`Error when parsing `rawPath: ${rawPath}`))` in line 160
for easier debugging?

Reply:
no, there's no error _thrown_. there's a result type of the error variant  _returned_ (we don't short circuit, control flow still as per normal). 

also, this is deleted and the updated code only retains the `errAsync` at the end! wrt debugging - actually we discard the error totally because we call `unwrapOr` at the use site, so whatever we put won't actually be seen. i'll still add it in but just an fyi!

what's happening atm is that i'm extending our data model to be more descriptive so rather than just parsing pages, we are also able to discern for eg, config/images/files, which might be more useful for devs as we can make more static guarantees rather than debugging post errors